peers/app_request_network.go

```diff
@@ -57,9 +56,20 @@ func NewNetwork(
 		),
 	)
 
+	// Create the info client
+	infoClient := info.NewClient(InfoAPINodeURL)
+	networkID, err := infoClient.GetNetworkID(context.Background())
+	if err != nil {
+		logger.Error(
+			"Failed to get network ID",
+			zap.Error(err),
+		)
+		return nil, nil, err
+	}
+
 	if networkID != constants.MainnetID &&
 		networkID != constants.FujiID &&
-		len(APINodeURL) == 0 {
+		InfoAPINodeURL == "" {
```

Comment:
I think `InfoAPINodeURL` being empty would cause an error above when it's used to get the network ID, so this case isn't possible. Do you know why it was required originally?

Reply:
Good catch, we will require this url be non-empty in all cases since it's used to fetch the network ID.

Looking at it again, this check is no longer valid. In an older release, the Avalanchego test network had a set of hard-coded node IDs for Fuji and mainnet, so we didn't need to explicitly connect to them here. That's no longer the case, so we need the info API to fetch the peers in all cases.